pkg/pillar/pubsub/publish.go

```diff
@@ -7,13 +7,15 @@ import (
 	"fmt"
 	"net"
 	"os"
+	"reflect"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/lf-edge/eve/pkg/pillar/types"
 	log "github.com/sirupsen/logrus"
 )
 
 // Usage:
-//  p1, err := pubsub.Publish("foo", fooStruct{})
+//  p1, err := pubsub.Publish("foo", fooStruct{}, parserFunc)
```

Comment:
Is this correct? As I understood it, both `Publication.Publish()` and `pubsub.Publish()` signatures are unchanged; they just add the parser and then call `pubsub.publishImpl()`, which is changed.

Reply:
Good catch. I had gone back and forth. Fixed now.